lib/using_restricted_array.rb

```diff
@@ -6,56 +6,120 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the array size
+# Space complexity: O(1)
 def length(array)
-  raise NotImplementedError
+  index = 0
+  until array[index] == nil
+    index += 1
+  end
+  return index
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is the array size
+# Space complexity: O(1)
 def print_array(array)
-  raise NotImplementedError
+  index = 0
+  until array[index] == nil
+    print array[index]
+    index += 1
+  end
 end
 
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(1)
```

Comment:
Since you loop through the array, and it depends on the size of the array.  It will be O(n)